### DISCUSSION TITLE
Wayland: Screen sharing with pipewire

### DISCUSSION THREAD

**Discussion by bodograumann** [qtile/qtile#3979] [Ideas]
So to get screen sharing working probably, I of course had to install several things, but most importantly I had to run the following:

```
dbus-update-activation-environment --systemd WAYLAND_DISPLAY XDG_CURRENT_DESKTOP=wlroots
systemctl --user stop pipewire pipewire-media-session xdg-desktop-portal xdg-desktop-portal-wlr
systemctl --user start pipewire-media-session
```

I would think that the first part should be done by default in qtile, shouldn't it?

**Comment by m-col** [comment]
I'd be happy for Qtile to do this if (and only if) Sway also does this, and using their same logic (i.e. regarding anything to do with checking systemd/dbus is running).

**Comment by clotodex** [comment]
I just have this in my autostart function in qtile (but yes it would be cool to maybe have it in the default config as well):
```python
    # setting environment for systemd and dbus
    denv = dict(os.environ)
    denv["XDG_CURRENT_DESKTOP"] = "qtile"
    p = subprocess.Popen(
        [
            "systemctl",
            "--user",
            "import-environment",
            "WAYLAND_DISPLAY",
            "XDG_CURRENT_DESKTOP",
        ],
        env=denv,
    ).wait()
```

**Comment by cr1ogen** [comment]
> So to get screen sharing working probably, I of course had to install several things, but most importantly I had to run the following:
> 
> ```
> dbus-update-activation-environment --systemd WAYLAND_DISPLAY XDG_CURRENT_DESKTOP=wlroots
> systemctl --user stop pipewire pipewire-media-session xdg-desktop-portal xdg-desktop-portal-wlr
> systemctl --user start pipewire-media-session
> ```
> 
> I would think that the first part should be done by default in qtile, shouldn't it?

Thanks for the info, I can now share screen in Qtile, but I only need your first command.

`dbus-update-activation-environment --systemd WAYLAND_DISPLAY XDG_CURRENT_DESKTOP=wlroots`

And from the second command I just wrote this.

`systemctl --user stop  xdg-desktop-portal xdg-desktop-portal-wlr`

and it worked perfect.
These two commands are added to the qtile autostart.sh file
